doc/nrf/device_guides/working_with_nrf/nrf52/developing.rst

```diff
@@ -82,6 +82,45 @@ If you want to add SMP Service to advertising data, refer to the :ref:`zephyr:sm
 
 .. fota_upgrades_over_ble_additional_information_end
 
+FOTA updates are also supported when MCUboot is in Direct-XIP mode.
+In this mode, the MCUboot bootloader boots an image directly from a given slot, so the swap operation is not needed.
+It can be used either with or without the revert mechanism support.
+For more information about Direct-XIP mode and the revert mechanism support, go to the :doc:`mcuboot:index-ncs` page.
+
+To use MCUboot in Direct-XIP mode together with FOTA updates, do the following:
+
+* Enable the :kconfig:option:`CONFIG_BOOT_DIRECT_XIP` Kconfig option in the MCUboot bootloader configuration.
+* Enable the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option to build application update images for both slots.
+  For more information, go to :ref:`app_build_mcuboot_output` page.
+
+   * This Kconfig option automatically enables the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` Kconfig option to inform the application about the chosen MCUboot mode of operation.
+     Thus it is not required to set it manually if the Direct-XIP mode without the revert mechanism support is used.
+
+* Optionally, if you want to enable the revert mechanism support, you also need to:
+
+   * Enable the :kconfig:option:`CONFIG_BOOT_DIRECT_XIP_REVERT` Kconfig option in the MCUboot bootloader configuration.
+   * Enable the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT` Kconfig option to inform the application about the chosen MCUboot mode of operation.
+
+Here is an example of how you can build for the :ref:`peripheral_lbs` sample with MCUboot in Direct-XIP mode without the revert mechanism support:
+
+.. parsed-literal::
+   :class: highlight
+
+    west build -b *build_target* -- -Dmcuboot_CONFIG_BOOT_DIRECT_XIP=y -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y -DCONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT=y
+
+Here is an example of how you can build for the :ref:`peripheral_lbs` sample with MCUboot in Direct-XIP mode and the revert mechanism support:
+
+.. parsed-literal::
+   :class: highlight
+
+    west build -b *build_target* -- -Dmcuboot_CONFIG_BOOT_DIRECT_XIP=y -Dmcuboot_CONFIG_BOOT_DIRECT_XIP_REVERT=y -DCONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_NCS_SAMPLE_MCUMGR_BT_OTA_DFU=y -DCONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT=y -DCONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT=y
+
+.. note::
+   When building the application for the first time with MCUboot in Direct-XIP mode and the revert mechanism support, you need to build the application with the additional option ``-DCONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS=\"--confirm\"``.
+   This option will confirm the application during the image signing process.
+   If the application is built without this option, it will fail to boot.
+   This option must be disabled when building update images.
```

Comment:
It would be great to generate the flashed hex (`zephyr.hex`) with this option enabled and the `app_update.bin` with this option disabled. I am not sure if our build system supports it. Changing project configuration when you want to flash the DK or generate the DFU zip diminishes the UX. We can leave it for now, but it looks like a major thing for a future improvement

Reply:
thanks